src/table/src/requests.rs

```diff
@@ -50,6 +50,17 @@ pub struct CreateTableRequest {
     pub engine: String,
 }
 
+pub const IMMUTABLE_TABLE_META_KEY: &str = "IMMUTABLE_TABLE_META";
+pub const IMMUTABLE_TABLE_LOCATION_KEY: &str = "LOCATION";
+pub const IMMUTABLE_TABLE_PATTERN_KEY: &str = "PATTERN";
+pub const IMMUTABLE_TABLE_FORMAT_KEY: &str = "FORMAT";
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(default)]
+pub struct ImmutableFileTableOptions {
+    pub files: Vec<String>,
+}
+
```

Comment:
As the table crate is the facade layer, it'd be better to put these in the file-table-engine crate.